### PR TITLE
SSH auth cluster + workspace lint config + plan-mode rule

### LINKED ISSUE #238: Does not use ssh-agent to retrieve passwords for ssh-keys
Upon trying to connect to a machine with `distant launch ssh://my.remote.ip.addr:port` I get the following message:
`Passphrase to decrypt /home/user/.ssh/key_file for user@my.remote.ip.addr:`
While using ssh directly does not prompt for the passphrase, because it correctly uses the running ssh-agent.

### LINKED ISSUE #251: Fails to authenticate using public key if host is named in .ssh/config
As the title says, it does not work if the Host is named in the client SSH config.

`~/.ssh/config`
```
Host nostromo
  HostName 192.168.1.3
  IdentityFile "~/path/to/key"
  User user
```

```sh
./distant connect ssh://nostromo \
        --options 'backend=libssh'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer

./distant connect ssh://user@nostromo \
        --options 'backend=libssh'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer

./distant connect ssh://user@192.168.1.3 \
        --options 'backend=libssh'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer
```
Same problem even when the path is provided using options.
```sh
./distant connect ssh://nostromo \
        --options 'backend=libssh,identity_files="~/path/to/key",identities_only=true'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer

./distant connect ssh://user@nostromo \
        --options 'backend=libssh,identity_files="~/path/to/key",identities_only=true'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer

./distant connect ssh://user@192.168.1.3 \
        --options 'backend=libssh,identity_files="~/path/to/key",identities_only=true'
Failed to connect to server

Caused by:
    Fatal: Fatal: Socket error: Connection reset by peer
```

It works as soon as I add a Host entry for the IP. After adding that, it works when I use the IP and the hostname from the other Host entry.
`~/.ssh/config`
```
Host nostromo
  HostName 192.168.1.3
  IdentityFile "~/path/to/key"
  User user

Host 192.168.1.3
  HostName 192.168.1.3
  IdentityFile "~/path/to/key"
  User user
```
```sh
./distant connect ssh://nostromo \
        --options 'backend=libssh'
Passphrase (/path/to/key):
4021718500

 took 3s
./distant connect ssh://192.168.1.3 \
        --options 'backend=libssh'
Passphrase (/path/to/key):
3707766213

./distant connect ssh://user@192.168.1.3 \
        --options 'backend=libssh'
Passphrase (/path/to/key):
4071324980
```

Trace for following command with only the first Host entry that uses its name.
```sh
./distant connect ssh://nostromo \
        --options 'backend=libssh' \
        --log-level trace \
        --log-file ./log.log
```
```
[2025-01-08 12:41:28.943342 +01:00] DEBUG [src/cli/commands/client.rs:65] Connecting to manager
[2025-01-08 12:41:28.943590 +01:00] DEBUG [distant-net/src/common/connection.rs:220] [Conn 3197600893] Waiting for server version
[2025-01-08 12:41:28.943701 +01:00] DEBUG [distant-net/src/common/connection.rs:231] [Conn 3197600893] Checking compatibility between client 0.20.0 & server 0.20.0
[2025-01-08 12:41:28.943712 +01:00] DEBUG [distant-net/src/common/connection.rs:244] [Conn 3197600893] Performing handshake
[2025-01-08 12:41:28.943727 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:705] [Handshake | Client] Waiting on options
[2025-01-08 12:41:28.946600 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:714] [Handshake | Client] Selecting from options: Options { compression_types: [Deflate, Gzip, Zlib], encryption_types: [XChaCha20Poly1305] }
[2025-01-08 12:41:28.946626 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:738] [Handshake | Client] Reporting choice: Choice { compression_level: None, compression_type: None, encryption_type: Some(XChaCha20Poly1305) }
[2025-01-08 12:41:28.946642 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:767] [Handshake | Client] Building compression & encryption codecs based on Choice { compression_level: None, compression_type: None, encryption_type: Some(XChaCha20Poly1305) }
[2025-01-08 12:41:28.946650 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:837] [Handshake | Client] Exchanging public key and salt
[2025-01-08 12:41:28.947267 +01:00] TRACE [distant-net/src/common/transport/framed.rs:849] [Handshake | Client] Waiting on public key and salt from other side
[2025-01-08 12:41:28.949456 +01:00] TRACE [distant-net/src/common/transport/framed.rs:860] [Handshake | Client] Deriving shared secret key
[2025-01-08 12:41:28.950447 +01:00] TRACE [distant-net/src/common/transport/framed.rs:794] [Handshake | Client] Bundling codecs
[2025-01-08 12:41:28.950460 +01:00] DEBUG [distant-net/src/common/connection.rs:249] [Conn 3197600893] Communicating that this is a new connection
[2025-01-08 12:41:28.950552 +01:00] DEBUG [distant-net/src/common/connection.rs:254] [Conn 3197600893] Receiving new connection id
[2025-01-08 12:41:28.950599 +01:00] DEBUG [distant-net/src/common/connection.rs:261] [Conn 3197600893] Resetting id to 2677813031
[2025-01-08 12:41:28.950607 +01:00] DEBUG [distant-net/src/common/connection.rs:266] [Conn 2677813031] Performing authentication
[2025-01-08 12:41:28.950613 +01:00] TRACE [distant-net/src/authentication.rs:69] Authenticate::authenticate waiting on next authentication frame
[2025-01-08 12:41:28.950641 +01:00] TRACE [distant-net/src/authentication.rs:72] Authenticate::Initialization(Initialization { methods: ["none"] })
[2025-01-08 12:41:28.950652 +01:00] TRACE [distant-net/src/authentication.rs:74] Writing data as frame: [130, 164, 116, 121, 112, 101, 188, 97, 117, 116, 104, 95, 105, 110, 105, 116, 105, 97, 108, 105, 122, 97, 116, 105, 111, 110, 95, 114, 101, 115, 112, 111, 110, 115, 101, 167, 109, 101, 116, 104, 111, 100, 115, 145, 164, 110, 111, 110, 101]
[2025-01-08 12:41:28.950687 +01:00] TRACE [distant-net/src/authentication.rs:69] Authenticate::authenticate waiting on next authentication frame
[2025-01-08 12:41:28.952935 +01:00] TRACE [distant-net/src/authentication.rs:99] Authenticate::StartMethod(StartMethod { method: "none" })
[2025-01-08 12:41:28.952952 +01:00] TRACE [distant-net/src/authentication.rs:69] Authenticate::authenticate waiting on next authentication frame
[2025-01-08 12:41:28.955368 +01:00] TRACE [distant-net/src/authentication.rs:103] Authenticate::Finished
[2025-01-08 12:41:28.955393 +01:00] DEBUG [distant-net/src/common/connection.rs:270] [Conn 2677813031] Deriving future OTP for reauthentication
[2025-01-08 12:41:28.955400 +01:00] DEBUG [distant-net/src/common/transport/framed.rs:837] Exchanging public key and salt
[2025-01-08 12:41:28.957223 +01:00] TRACE [distant-net/src/common/transport/framed.rs:849] Waiting on public key and salt from other side
[2025-01-08 12:41:28.957249 +01:00] TRACE [distant-net/src/common/transport/framed.rs:860] Deriving shared secret key
[2025-01-08 12:41:28.958010 +01:00] INFO [distant-net/src/common/connection.rs:273] [Conn 2677813031] Connect completed successfully!
[2025-01-08 12:41:28.958066 +01:00] INFO [src/cli/common/client.rs:79] Connected to unix socket @ "/run/user/1000/distant/user.distant.sock"
[2025-01-08 12:41:28.958079 +01:00] DEBUG [src/cli/commands/client.rs:69] Connecting to server at ssh://nostromo with backend="libssh"
[2025-01-08 12:41:28.958089 +01:00] TRACE [distant-net/src/manager/client.rs:133] connect(ssh://nostromo, backend="libssh")
[2025-01-08 12:41:28.958205 +01:00] TRACE [distant-net/src/client.rs:357] Client sending ��id�6808163447508027682�payload��type�connect�destination�ssh://nostromo�options�backend="libssh"
[2025-01-08 12:41:28.965570 +01:00] TRACE [distant-net/src/client.rs:301] Client receiving (id:16414558784599284863 | origin: 6808163447508027682): ��type�authenticate�id��av��msg��type�auth_info�text�1Using libssh-rs to connect to user@192.168.1.3:22
[2025-01-08 12:41:28.965722 +01:00] TRACE [distant-net/src/client.rs:324] Client delivered response 16414558784599284863 to 6808163447508027682
[2025-01-08 12:41:28.965769 +01:00] INFO [distant-net/src/manager/client.rs:188] Using libssh-rs to connect to user@192.168.1.3:22
[2025-01-08 12:41:29.023091 +01:00] TRACE [distant-net/src/client.rs:301] Client receiving (id:15356765438256929507 | origin: 6808163447508027682): ��type�authenticate�id�^V��^�msg��type�auth_info�text�SSH-2.0-OpenSSH_9.8
[2025-01-08 12:41:29.023136 +01:00] TRACE [distant-net/src/client.rs:324] Client delivered response 15356765438256929507 to 6808163447508027682
[2025-01-08 12:41:29.023158 +01:00] INFO [distant-net/src/manager/client.rs:188] SSH-2.0-OpenSSH_9.8
[2025-01-08 12:41:29.063565 +01:00] TRACE [distant-net/src/client.rs:301] Client receiving (id:712675925819622796 | origin: 6808163447508027682): ��type�authenticate�idΦKn^V�msg��type�auth_error�kind�fatal�text�Dunhandled auth case; methods=PUBLIC_KEY, status={PUBLIC_KEY: Denied}
[2025-01-08 12:41:29.063598 +01:00] TRACE [distant-net/src/client.rs:324] Client delivered response 712675925819622796 to 6808163447508027682
[2025-01-08 12:41:29.063617 +01:00] TRACE [distant-net/src/client.rs:301] Client receiving (id:13709783148468818300 | origin: 6808163447508027682): ��type�error�description�Dunhandled auth case; methods=PUBLIC_KEY, status={PUBLIC_KEY: Denied}
[2025-01-08 12:41:29.063624 +01:00] TRACE [distant-net/src/client.rs:324] Client delivered response 13709783148468818300 to 6808163447508027682
[2025-01-08 12:41:29.063622 +01:00] ERROR [distant-net/src/manager/client.rs:191] unhandled auth case; methods=PUBLIC_KEY, status={PUBLIC_KEY: Denied}
[2025-01-08 12:41:29.064041 +01:00] ERROR [src/lib.rs:143] Failed to connect to server

Caused by:
    Fatal: unhandled auth case; methods=PUBLIC_KEY, status={PUBLIC_KEY: Denied}
```

### LINKED ISSUE #252: Does not use keys from agent or from "identity_files" option
To use a key, I have to explicitly set it in `~/.ssh/.config` using `IdentityFile`. It would be nice if it got the key from the running SSH agent instead! 

I tried using the `identity_files` option, but it still won't work. Not sure if something is broken with my installation or if it's bugged?

For reference,
```sh
./distant connect ssh://user@192.168.1.3 \
        --options 'backend=libssh,identity_files="~/path/to/key",identities_only=true'
Failed to connect to server

Caused by:
    Fatal: unhandled auth case; methods=PUBLIC_KEY, status={PUBLIC_KEY: Denied}
```

I shared a trace log from another issue I added just now, https://github.com/chipsenkbeil/distant/issues/251. The command was ran when my agent was running, so if it used the agent it should've gotten the key.

### PR DESCRIPTION
**Pull request by chipsenkbeil**
## Summary

- **TODO.md audit**: Remove 8 resolved items, update 4 stale descriptions, add regression test for #162 (known_hosts with whitespace paths)
- **SSH agent authentication** (#238, #252): Try agent keys before file-based auth. Unix via `SSH_AUTH_SOCK`, Windows via OpenSSH named pipe + Pageant fallback. Respects `identities_only` option.
- **Tilde expansion** (#251, #252): Expand `~` in CLI-provided identity file paths (ssh2-config already handles config-sourced paths)
- **Passphrase prompting** (#238): Prompt for passphrase when a key file is encrypted instead of silently skipping it. Works in both CLI and plugin/neovim contexts.
- **SSH auth refactoring**: Extract auth logic into dedicated module with trait-based abstractions for testability. Polish naming (`HandleAgent` → `AgentHandle`), use pattern matching for encrypted key detection, clean up doc comments and inline comments.
- **Test cleanup**: Remove 60 low-value lib.rs tests, rename survivors to naming convention, add mock-based unit tests for auth module.
- **Workspace lint config**: Replace `RUSTFLAGS="-Dwarnings"` with `[workspace.lints.clippy] all = deny` and `[workspace.lints.rust] warnings = deny` in root Cargo.toml. All 6 crates inherit via `[lints] workspace = true`. CI/release workflows, agent definitions, and CLAUDE.md updated accordingly.
- **Plan-mode agent rule**: Add "Plan-Mode Requirements" section to CLAUDE.md requiring every plan to declare agent usage, execution order, and justification for skipped stages.

Closes #238, closes #251, closes #252

## Test plan

- [x] Unit tests for `expand_tilde` (5 tests: home expansion, absolute/relative passthrough, bare tilde, idempotency)
- [x] Regression test for #162: known_hosts lookup and TOFU learning through whitespace-containing directory paths
- [x] Mock-based unit tests for auth module (agent auth, keyboard-interactive, password auth)
- [x] All distant-ssh unit + integration tests pass
- [x] `cargo clippy --all-features --workspace --all-targets` passes without RUSTFLAGS (workspace lints working)
- [x] `cargo test --all-features -p distant-core` passes (2087 tests)
- [x] Zero `RUSTFLAGS.*Dwarnings` references remain in repo
- [ ] Manual test: `distant ssh <config-alias>` with HostName + IdentityFile in ~/.ssh/config
- [ ] Manual test: load passphrase-protected key in ssh-agent, verify `distant ssh` uses it without prompting
- [ ] CI: verify Windows agent code compiles (behind `#[cfg(windows)]`)

### CODE REVIEW

File: distant-ssh/src/auth.rs [hunk 1]
```diff
@@ -0,0 +1,2084 @@
...
+use russh::client::{AuthResult, KeyboardInteractiveAuthResponse};
+use russh::keys::PrivateKeyWithHashAlg;
+use russh::keys::agent::client::AgentClient;
+
+use crate::{ClientHandler, SshAuthEvent, SshAuthHandler, SshAuthPrompt};
+
+/// Abstraction over SSH session authentication operations.
+///
+/// Implemented by `russh::client::Handle<ClientHandler>` for production use.
+/// A mock implementation is provided in tests for per-code-path unit testing.
```
Comment (chipsenkbeil) [inline distant-ssh/src/auth.rs]: Unneeded to cite an implementation detail like this that is situational.

File: distant-ssh/src/lib.rs [hunk 2]
```diff
@@ -859,221 +850,79 @@ impl Ssh {
...
-                        ssh_dir.join("id_rsa"),
-                        ssh_dir.join("id_ecdsa"),
-                    ];
-                    defaults.into_iter().filter(|p| p.exists()).collect()
-                } else {
-                    warn!("Could not determine home directory; skipping default key discovery");
-                    Vec::new()
-                }
-            };
+        // Agent auth
```
Comment (chipsenkbeil) [inline distant-ssh/src/lib.rs]: Provide a more meaningful comment

File: distant-ssh/src/lib.rs [hunk 3]
```diff
@@ -859,221 +850,79 @@ impl Ssh {
...
+                {
+                    self.authenticated = true;
+                    return Ok(());
                 }
             }
         }
 
-        // --- Keyboard-interactive authentication ---
-        // Track whether we already prompted the user (to avoid double-prompting with password)
+        // Keyboard-interactive
```
Comment (chipsenkbeil) [inline distant-ssh/src/lib.rs]: The former comment mentioning double-prompting seems more meaningful versus the new comment.
